fix(chat/notes): resolve persistent visual notifications in notes and chat hooks

Arthurk12 opened the pull request:
### What does this PR do?
- For notes: markNotesAsRead was not executed when isVisible was true (always true while in notes panel, i.e. not pinned). Moved it inside the pad and linked to iframe load to improve the relationship between
iframe loading and marking as viewed.
- Both notes and chat hooks had issues recalculating when there were certain updates in the last read timestamp. The latch logic was wrong and prevented the hooks from recalculating internal states. Simplified the latch logic in both hooks using memo for reactive computation.

- Additionally, fixes an incorrectly open chat id logic

### Closes Issue(s)
Closes #24255

Arthurk12 committed:
fix(chat): incorrectly setting open chat id

Arthurk12 committed:
fix(hooks): resolve persistent visual notifications in notes and chat
hooks

- For notes: markNotesAsRead was not executed when isVisible was true
(always true while in notes panel, i.e. not pinned). Moved it inside
the pad and linked to iframe load to improve the relationship between
iframe loading and marking as viewed.
- Both notes and chat hooks had issues recalculating when there were
certain updates in the last read timestamp. The latch logic was wrong
and prevented the hooks from recalculating internal states. Simplified
the latch logic in both hooks using memo for reactive computation.

germanocaumo reviewed (changes requested):
I tested it but for me it's still showing the notification dot wrongly sometimes (not always)

Arthurk12 committed:
fix(shared-notes): correct marking as read

Adjust the condition to mark notes as read, as they have a special
unmount delay logic. So the mark as read function can rely only on
component unmount as it may happen `delayForUnmountOfSharedNote` ms
after the shared notes is effectively closed.

germanocaumo reviewed (approved):
LGTM
